pkg/manifest/manifest_test.go

```diff
@@ -155,6 +161,76 @@ data:
 
 }
 
+func TestParseManifestsDuplicates(t *testing.T) {
+	tests := []struct {
+		name    string
+		raw     string
+		wantErr string
+	}{{
+		name: "no-duplicate",
+		raw: `
+apiVersion: extensions/v1
+kind: ConfigMap
+metadata:
+  name: a-config
+  namespace: default
+---
+apiVersion: extensions/v1
+kind: ConfigMap
+metadata:
+  name: b-config
+  namespace: default
+`,
+		wantErr: "",
+	}, {
+		name: "resource-id-error",
+		raw: `
+apiVersion: extensions/v1
+kind: Kind
+metadata:
+  name:
+  namespace: default
+`,
+		wantErr: "must contain kubernetes required fields kind and name",
+	}, {
+		name: "duplicate",
+		raw: `
+apiVersion: extensions/v1
+kind: ConfigMap
+metadata:
+  name: a-config
+  namespace: default
+---
+apiVersion: extensions/v2
+kind: ConfigMap
+metadata:
+  name: a-config
+  namespace: default
+`,
+		wantErr: "duplicate resource:",
```

Comment:
should we grow this to ensure the explanation of the duplicate includes the description we want?  Looks like it will be `duplicate resource: (Group: "extensions" Kind: "ConfigMap" Namespace: "default" Name: "a-config")`

Reply:
done